src/attiny_rtc_i2c.rs

```diff
@@ -727,4 +727,131 @@ impl SharedI2C {
     pub fn tell_attiny_to_power_down_rp2040(&mut self, delay: &mut Delay) -> Result<(), Error> {
         self.try_attiny_write_command(REG_RP2040_PI_POWER_CTRL, 0x02, delay)
     }
+
+    fn attiny_read_eeprom_command(&mut self, command: u8, payload: &mut [u8]) -> Result<(), Error> {
+        let lock_pin = self.unlocked_pin.take().unwrap();
+        let is_low = lock_pin.is_low().unwrap_or(false);
+        if is_low {
+            let pin = lock_pin.into_pull_type::<PullUp>();
+            let result = self.i2c().write_read(EEPROM_ADDRESS, &[command], payload);
+            self.unlocked_pin = Some(pin.into_pull_type::<PullDown>());
+            return result;
+        } else {
+            self.unlocked_pin = Some(lock_pin);
+            Err(Error::Abort(1))
+        }
+    }
+
+    fn try_attiny_read_eeprom_command(
```

Comment:
Probably shouldn't be called `try_attiny_read_eeprom_command`, since no attiny is involved here?